groups/bsl/bslstl/bslstl_optional.h

```diff
@@ -765,787 +1432,11454 @@ void Optional_DataImp<TYPE>::emplace(
+            BSLS_COMPILERFEATURES_FORWARD(ARGS_05, args_05),
+            BSLS_COMPILERFEATURES_FORWARD(ARGS_06, args_06),
+            BSLS_COMPILERFEATURES_FORWARD(ARGS_07, args_07),
+            BSLS_COMPILERFEATURES_FORWARD(ARGS_08, args_08),
+            BSLS_COMPILERFEATURES_FORWARD(ARGS_09, args_09));
+}
+#endif  // BSLSTL_OPTIONAL_VARIADIC_LIMIT_J >= 9
+
+#if BSLSTL_OPTIONAL_VARIADIC_LIMIT_J >= 10
+template <class TYPE, bool USES_BSLMA_ALLOC>
+template <class ARGS_01,
+          class ARGS_02,
+          class ARGS_03,
+          class ARGS_04,
+          class ARGS_05,
+          class ARGS_06,
+          class ARGS_07,
+          class ARGS_08,
+          class ARGS_09,
+          class ARGS_10>
+inline
+optional<TYPE, USES_BSLMA_ALLOC>::optional(
+                               bsl::in_place_t,
+                            BSLS_COMPILERFEATURES_FORWARD_REF(ARGS_01) args_01,
+                            BSLS_COMPILERFEATURES_FORWARD_REF(ARGS_02) args_02,
+                            BSLS_COMPILERFEATURES_FORWARD_REF(ARGS_03) args_03,
+                            BSLS_COMPILERFEATURES_FORWARD_REF(ARGS_04) args_04,
+                            BSLS_COMPILERFEATURES_FORWARD_REF(ARGS_05) args_05,
+                            BSLS_COMPILERFEATURES_FORWARD_REF(ARGS_06) args_06,
+                            BSLS_COMPILERFEATURES_FORWARD_REF(ARGS_07) args_07,
+                            BSLS_COMPILERFEATURES_FORWARD_REF(ARGS_08) args_08,
+                            BSLS_COMPILERFEATURES_FORWARD_REF(ARGS_09) args_09,
+                            BSLS_COMPILERFEATURES_FORWARD_REF(ARGS_10) args_10)
+{
+    emplace(BSLS_COMPILERFEATURES_FORWARD(ARGS_01, args_01),
+            BSLS_COMPILERFEATURES_FORWARD(ARGS_02, args_02),
+            BSLS_COMPILERFEATURES_FORWARD(ARGS_03, args_03),
+            BSLS_COMPILERFEATURES_FORWARD(ARGS_04, args_04),
+            BSLS_COMPILERFEATURES_FORWARD(ARGS_05, args_05),
+            BSLS_COMPILERFEATURES_FORWARD(ARGS_06, args_06),
+            BSLS_COMPILERFEATURES_FORWARD(ARGS_07, args_07),
+            BSLS_COMPILERFEATURES_FORWARD(ARGS_08, args_08),
+            BSLS_COMPILERFEATURES_FORWARD(ARGS_09, args_09),
+            BSLS_COMPILERFEATURES_FORWARD(ARGS_10, args_10));
+}
+#endif  // BSLSTL_OPTIONAL_VARIADIC_LIMIT_J >= 10
+
+
+#if defined(BSLS_COMPILERFEATURES_SUPPORT_GENERALIZED_INITIALIZERS)
+#if BSLSTL_OPTIONAL_VARIADIC_LIMIT_J >= 0
+template <class TYPE, bool USES_BSLMA_ALLOC>
+template <class INIT_LIST_TYPE>
+inline
+optional<TYPE, USES_BSLMA_ALLOC>::optional(
+                               bsl::in_place_t,
+                               std::initializer_list<INIT_LIST_TYPE>      il)
+{
+    emplace(il);
+}
+#endif  // BSLSTL_OPTIONAL_VARIADIC_LIMIT_J >= 0
+
+#if BSLSTL_OPTIONAL_VARIADIC_LIMIT_J >= 1
+template <class TYPE, bool USES_BSLMA_ALLOC>
+template <class INIT_LIST_TYPE, class ARGS_01>
+inline
+optional<TYPE, USES_BSLMA_ALLOC>::optional(
+       
```

Comment:
Just as a note to comment above - here `this->` is just nice for symmetry (unlike the constructors)

Reply:
i can't see where you mean as git is being silly and not showing the correct line. Can you tell me which method you are looking at ?

Reply:
This is next to `value_or` I think.

Reply:
would you like it back in all the places in `value_or` for symmetry ? `value_or` is where we've decided to keep it because of the `value` argument.

Reply:
If there's a clash - and `this->` solves it, I think it's fine.

Reply:
I'm trying to understand the symmetry point above. What I meant to ask is if the symmetry means that all calls should be qualified with `this->` in a `value_or` method ? 
For example 
current value_or
`template <class TYPE, bool USES_BSLMA_ALLOC>`
`template <class ANY_TYPE>`
`inline`
`TYPE optional<TYPE, USES_BSLMA_ALLOC>::value_or(ANY_TYPE&& value) &&`
`{`
`    if (has_value()) {`
`       return TYPE(std::move(this->value()));                        // Nina : this needed for name resolution here`
`    }`
`    else {`
`        return TYPE(std::forward<ANY_TYPE>(value));                   // RETURN`
`    }`
`}`


would you like to modify above to
`template <class TYPE, bool USES_BSLMA_ALLOC>`
`template <class ANY_TYPE>`
`inline`
`TYPE optional<TYPE, USES_BSLMA_ALLOC>::value_or(ANY_TYPE&& value) &&`
`{`
`    if (this->has_value()) {                       // this added here for symmetry ?`
`        return TYPE(std::move(this->value()));                        `
`    }`
`    else {`
`        return TYPE(std::forward<ANY_TYPE>(value));                   `
`    }`
`}`

Reply:
I mean this code:
```
template <class TYPE, bool USES_BSLMA_ALLOC>
void optional<TYPE, USES_BSLMA_ALLOC>::swap(optional& other)
{
    if (this->has_value() && other.has_value()) {
        BloombergLP::bslalg::SwapUtil::swap(
                                           BSLS_UTIL_ADDRESSOF(this->value()),
                                           BSLS_UTIL_ADDRESSOF(other.value()));
    }
    else if (this->has_value()) {
        other.emplace(MoveUtil::move(this->value()));
        this->reset();
    }
    else if (other.has_value()) {
        this->emplace(MoveUtil::move(other.value()));
        other.reset();
    }
}
```
where `this->value()` and `other.value()` line up nicely.

Reply:
ah, yes. I see. I do agree it looks good there.
I've added `this->` back to `swap` functions. Let me know what you think.

Reply:
All good, let's close this.